### PR TITLE
support UniqueWithCounts for TF FE

### LINKED ISSUE #20618: [Good First Issue][TF FE]: Support UniqueWithCounts operation for TensorFlow models
### Context

OpenVINO component responsible for support of TensorFlow models is called as TensorFlow Frontend (TF FE). TF FE converts a model represented in [TensorFlow opset](https://www.tensorflow.org/api_docs/python/tf/raw_ops) to a model in [OpenVINO opset](https://docs.openvino.ai/2023.1/openvino_docs_ops_opset.html).

In order to infer TensorFlow models with [UniqueWithCounts](https://www.tensorflow.org/api_docs/python/tf/raw_ops/UniqueWithCounts) operation by OpenVINO, TF FE needs to be extended with this operation support.

### What needs to be done?

For [UniqueWithCounts](https://www.tensorflow.org/api_docs/python/tf/raw_ops/UniqueWithCounts) operation support, you need to add the corresponding loader into TF FE [op](https://github.com/openvinotoolkit/openvino/tree/master/src/frontends/tensorflow_common/src/op) directory and to register it into the dictionary of [Loaders](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/tensorflow/src/op_table.cpp). One loader is responsible for conversion (or decomposition) of one type of TensorFlow operation.

Here is an example of loader implementation for TensorFlow `Einsum` operation:

```cpp
OutputVector translate_einsum_op(const NodeContext& node) { 
     auto op_type = node.get_op_type(); 
     TENSORFLOW_OP_VALIDATION(node, op_type == "Einsum", "Internal error: incorrect usage of translate_einsum_op."); 
     auto equation = node.get_attribute<std::string>("equation"); 
  
     OutputVector inputs; 
     for (size_t input_ind = 0; input_ind < node.get_input_size(); ++input_ind) { 
         inputs.push_back(node.get_input(input_ind)); 
     } 
  
     auto einsum = make_shared<Einsum>(inputs, equation); 
     set_node_name(node.get_name(), einsum); 
     return {einsum}; 
 } 
```

In this example, `translate_einsum_op` converts TF `Einsum` into OV `Einsum`. `NodeContext` object passed into the loader packs all info about inputs and attributes of `Einsum` operation. The loader retrieves an attribute of the equation by using the `NodeContext::get_attribute()` method, prepares input vector, creates `Einsum` operation from OV opset and returns a vector of outputs.

Responsibility of a loader is to parse operation attributes, prepare inputs and express TF operation via OV operations sub-graph. Example for `Einsum` demonstrates the resulted sub-graph with one operation. In PR https://github.com/openvinotoolkit/openvino/pull/19007 you can see operation decomposition into multiple node sub-graph.

### Hint

Use [Unique](https://docs.openvino.ai/2023.1/openvino_docs_ops_movement_Unique_10.html) operation from OV opset that provides all necessary outputs: unique elements, indices, and counts. :wink:

### Example Pull Requests

- https://github.com/openvinotoolkit/openvino/pull/19007

### Resources

- [What is OpenVINO?](https://github.com/openvinotoolkit/openvino#what-is-openvino-toolkit)
- [OpenVINO TensorFlow Frontend](https://github.com/openvinotoolkit/openvino/blob/master/src/frontends/tensorflow/README.md)
- [Contribution guide](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [Blog post on contributing to OpenVINO](https://github.com/openvinotoolkit/openvino/blob/master/CONTRIBUTING.md)
- [User documentation](https://docs.openvino.ai/)

### Contact points

@openvinotoolkit/openvino-tf-frontend-maintainers 

### Ticket

TBD

### PR DESCRIPTION
**Pull request by ahmadchalhoub**
### Details:
Extend UniqueWithCounts (https://www.tensorflow.org/api_docs/python/tf/raw_ops/UniqueWithCounts) operator for OpenVino TF FE.

### Tickets:
- Closes #20618

### CODE REVIEW

File: src/frontends/tensorflow_common/src/op/unique_with_counts.cpp [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+
+OutputVector translate_unique_with_counts_op(const NodeContext& node) {
+    default_op_checks(node, 1, {"UniqueWithCounts"});
+
+    // get input 'x' from node and node name
+    auto x = node.get_input(0);
+    auto node_name = node.get_name();
+
+    auto unique = make_shared<v10::Unique>(x, false, ov::element::i32, ov::element::i32);
```
Comment (popovaan) [inline src/frontends/tensorflow_common/src/op/unique_with_counts.cpp:24]: Please get the name from `node` and set it for this new operation. Here is example from another translator: https://github.com/openvinotoolkit/openvino/blob/e2e9d6056f066e77722265e17c81a7071e0804c7/src/frontends/tensorflow_common/src/op/fft.cpp#L53

**Comment by popovaan** [comment]
Please fix code style. 
You can do it easily using following commands:
```
cmake -DENABLE_CLANG_FORMAT=ON ..
make clang_format_fix_all
```

File: tests/layer_tests/tensorflow_tests/test_tf_UniqueWithCounts.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+# SPDX-License-Identifier: Apache-2.0
+
+import numpy as np
+import pytest
+import tensorflow as tf
+from common.tf_layer_test_class import CommonTFLayerTest
+
+
+class TestUnique(CommonTFLayerTest):
+    def _prepare_input(self, inputs_info):
```
Comment (popovaan) [inline tests/layer_tests/tensorflow_tests/test_tf_UniqueWithCounts.py:11]: You don't need to override `_prepare_input` here. CommonLayerTest class already has this method, which can be used in most cases: https://github.com/openvinotoolkit/openvino/blob/e2e9d6056f066e77722265e17c81a7071e0804c7/tests/layer_tests/common/layer_test_class.py#L149

We need to ovverride _prepare_input only for complex type.

File: src/frontends/tensorflow_common/src/op/unique_with_counts.cpp (hunk 1 above)
Comment (ahmadchalhoub) [reply src/frontends/tensorflow_common/src/op/unique_with_counts.cpp:23]: @popovaan, thanks for the feedback! Just pushed requested changes.

**Review by popovaan — APPROVED** [review]

**Review by mvafin — APPROVED** [review]

**Comment by mvafin** [comment]
build_jenkins

**Comment by p-wysocki** [comment]
Thank you for your contribution @ahmadchalhoub! 

Are we ready to merge this? CI is green, there are 2 approves.

**Comment by andrei-kochin** [comment]
@p-wysocki  Done!
@ahmadchalhoub Confratulations!

**Pull request merged**